controllers/common.go

```diff
@@ -118,30 +157,63 @@ func ValidateResticSecret(resticsecret *corev1.Secret) error {
 	if resticsecret.Data == nil {
 		return errors.New("secret data is empty")
 	}
-	for key, val := range resticsecret.Data {
-		switch key {
-		case AWSAccessKey:
-			b := checkByteArrayIsEmpty(val)
-			if !b {
-				return errors.New("awsAccessKey value cannot be empty")
-			}
-		case AWSSecretKey:
-			b := checkByteArrayIsEmpty(val)
-			if !b {
-				return errors.New("awsSecretKey value cannot be empty")
-			}
-		case ResticPassword:
-			b := checkByteArrayIsEmpty(val)
-			if !b {
-				return errors.New("resticPassword value cannot be empty")
+
+	provider := resticsecret.Labels[OADPBSLProviderName]
+
+	switch provider {
+	case AWSProvider:
+		for key, val := range resticsecret.Data {
+			switch key {
+			case AWSAccessKey:
+				b := checkByteArrayIsEmpty(val)
+				if !b {
+					return errors.New("awsAccessKey value cannot be empty")
+				}
+			case AWSSecretKey:
+				b := checkByteArrayIsEmpty(val)
+				if !b {
+					return errors.New("awsSecretKey value cannot be empty")
+				}
+			case ResticPassword:
+				b := checkByteArrayIsEmpty(val)
+				if !b {
+					return errors.New("resticPassword value cannot be empty")
+				}
+			case ResticRepository:
+				b := checkByteArrayIsEmpty(val)
+				if !b {
+					return errors.New("resticRepository value cannot be empty")
+				}
 			}
-		case ResticRepository:
-			b := checkByteArrayIsEmpty(val)
-			if !b {
-				return errors.New("resticRepository value cannot be empty")
+		}
+
+	case AzureProvider:
+		for key, val := range resticsecret.Data {
+			switch key {
+			case AzureAccountName:
+				b := checkByteArrayIsEmpty(val)
+				if !b {
+					return errors.New("azure accout name value cannot be empty")
+				}
+			case AzureAccountKey:
+				b := checkByteArrayIsEmpty(val)
+				if !b {
+					return errors.New("azure account key value cannot be empty")
+				}
+			case ResticPassword:
+				b := checkByteArrayIsEmpty(val)
+				if !b {
+					return errors.New("resticPassword value cannot be empty")
+				}
+			case ResticRepository:
+				b := checkByteArrayIsEmpty(val)
+				if !b {
+					return errors.New("resticRepository value cannot be empty")
+				}
 			}
 		}
 	}
+
```

Comment:
- Add a case for empty provider labe
- Add a default case for unsupported providers

Reply:
@kaovilai we do not support empty provider or any other provider other than aws/azure/gcp, its already validated in bsl.go https://github.com/openshift/oadp-operator/blob/master/controllers/bsl.go#L61

Reply:
Are we absolutely positive that no one else will be playing around with labels?

I see that we like to have error handling done in one place. Sometimes that's refactored away and then you'll hit issues.

Reply:
We've merged with assumptions of a prior validation in another function before so I'm not opposed to continuing the trend.. but it just itches me a little bit.